<?xml version="1.0"?>

<launch>


<rosparam>
  NUM_OF_ROBOTS: 3
  EXP_NAME: "exp35"
  DIST_INTO: 20.5
  DIST_LEAVE: 30.0
</rosparam>


<node pkg = "stage_ros" name = "stageros" type = "stageros" output = "screen" args="-d $(find distributed)/worlds/map_35_3r.world">
</node>

<node pkg = "rviz" name = "rviz" type = "rviz" args="-d $(find distributed)/rviz/rviz_config_3r.rviz" output="screen">
</node>

<node pkg = "distributed" name = "config_node" type = "config_3r.py" output="screen" args="9">
</node>

<node pkg = "map_server" name = "map_server" type = "map_server" args="$(find distributed)/maps/map_35_GVD.yaml" output="screen">
</node>


<!--node pkg = "distributed" name = "launcher" type = "launch_robots.py" args="3 ">
</node-->
<!--node pkg = "distributed" name = "follow_graph_0" type = "Algorithm_1.py" args="0" output="screen" launch-prefix="xterm -e">
</node-->
<node pkg = "distributed" name = "follow_graph_0" type = "Algorithm_1.py" args="0" output="screen" launch-prefix="xterm -hold -e">
</node>
<node pkg = "distributed" name = "follow_graph_1" type = "Algorithm_1.py" args="1" output="screen" launch-prefix="xterm -hold -e">
</node>
<node pkg = "distributed" name = "follow_graph_2" type = "Algorithm_1.py" args="2" output="screen" launch-prefix="xterm -hold -e">
</node>

<node pkg = "distributed" name = "sensor_simulator" type = "sensor_simulator.py" output="screen">
</node>

<!--SPS_ids = [0  1  2 3  4  5  6 7  8  9  10 11 12 13 14 15 16 17 18 19] this goes as arg for config_node-->
<!--SPS     = [41 69 6 16 50 25 8 44 34 55 51 74 43 24 57 13 18 17 31 71] -->
<node pkg = "distributed" name = "time_couter" type = "Count_time.py" args="37" output="screen">
</node>

<!--node pkg="tf" type="static_transform_publisher" name="base_to_laser_broadcaster_0" args="0 0 0 0 0 0 world robot_0/odom 10" /-->
<!--node pkg="tf" type="static_transform_publisher" name="base_to_laser_broadcaster_1" args="0 0 0 0 0 0 world robot_1/odom 10" /-->
<!--node pkg="tf" type="static_transform_publisher" name="base_to_laser_broadcaster_2" args="0 0 0 0 0 0 world robot_2/odom 10" /-->

<node pkg = "distributed" name = "histogram" type = "histVisitation.py" args="3 39" output="screen">
</node>


</launch>
